### PR TITLE
mes-5346 & mes-5508 Advanced search filters

### PR DESCRIPTION
**Pull request by RLCorp**
## Description
Added activity code and category filters for advanced search
## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

## Screenshots (optional)
image

### CODE REVIEW

File: src/app/pages/test-results-search/components/advanced-search/advanced-search.ts [hunk 1]
```diff
@@ -22,6 +24,53 @@ export class AdvancedSearchComponent {
...
+    TestCategory.EUA2M2,
+    TestCategory.EUAM1,
+    TestCategory.EUAM2,
+    TestCategory.EUAMM1,
+    TestCategory.EUAMM2,
+    TestCategory.F,
+    TestCategory.G,
+    TestCategory.H,
+    TestCategory.K,
+    TestCategory.SC,
```
Comment (matthew2564) [inline src/app/pages/test-results-search/components/advanced-search/advanced-search.ts]: Can we remove the following categories please - they don't exist in DES yet

A, A1, A2, ADI3, AM , B1, SC

File: src/app/providers/search/search.models.ts [hunk 2]
```diff
@@ -4,4 +4,6 @@ export interface AdvancedSearchParams {
   staffNumber?: string;
   costCode?: string;
   excludeAutoSavedTests?: string;
+  activityFilter?: string
+  categoryFilter?: string
```
Comment (matthew2564) [inline src/app/providers/search/search.models.ts]: Can we rename these to be ```activityCode``` and ```category``` please

File: src/app/pages/test-results-search/components/advanced-search/advanced-search.ts [hunk 3]
```diff
@@ -70,6 +119,15 @@ export class AdvancedSearchComponent {
...
   };
 
+  populateActivityArray() {
+    const activityCodeArray: any[] = [];
+    for (let i = 0; i < activityCodeModelList.length; i += 1) {
+      activityCodeArray.push(`${activityCodeModelList[i].activityCode} - ${activityCodeModelList[i].description}`);
+    }
+    activityCodeArray.unshift('All');
+    return activityCodeArray;
+  }
```
Comment (matthew2564) [inline src/app/pages/test-results-search/components/advanced-search/advanced-search.ts]: I don't think we need to do this - we already have an array of codes/descriptions of the codes, believe we could just do something like this
```
  activityCodes: { activityCode: string; description: string; }[] = [
    { activityCode: '', description: 'All' },
    ...activityCodeModelList,
  ];
```

File: src/app/pages/test-results-search/components/advanced-search/advanced-search.ts [hunk 4]
```diff
@@ -22,6 +24,53 @@ export class AdvancedSearchComponent {
   @Output()
   onSearchTests = new EventEmitter<AdvancedSearchParams>();
 
+  activityCodes: any[] = this.populateActivityArray();
+  testCategories: any[] = [
```
Comment (matthew2564) [inline src/app/pages/test-results-search/components/advanced-search/advanced-search.ts]: This can be typed to be a ```string[]```

**Review by matthew2564 — APPROVED** [review]

**Review by andrewsetterfield — APPROVED** [review]

**Pull request merged**